### ISSUE TITLE
IDE0067 ‘Disposable object is never disposed’ treats the last parameter differently

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#37528]
**Version Used**: VS 16.2.0

There should be no diagnostics reported here:

```cs
class C
{
    IDisposable M()
    {
        return CreateContainer(new[]
        {
            // ↓ IDE0067 Disposable object created by 'new DisposableThing()' is never disposed
            new DisposableThing(),
            // ↓ IDE0067 Disposable object created by 'new DisposableThing()' is never disposed
            new DisposableThing(),
            // No diagnostic, as expected
            new DisposableThing()
        });
    }

    IDisposable CreateContainer(DisposableThing[] items)
    {
        return new DisposableContainer(items);
    }

    class DisposableThing : IDisposable
    {
        public void Dispose()
        {
            throw new NotImplementedException();
        }
    }

    class DisposableContainer : IDisposable
    {
        public DisposableContainer(DisposableThing[] items)
        {
            Items = items;
        }

        public DisposableThing[] Items { get; }

        public void Dispose()
        {
            foreach (var item in Items)
                item.Dispose();
        }
    }
}
```

Same thing also with individual parameters (as well as params arrays):

```cs
IDisposable M()
{
    return CreateContainer(
        // ↓ IDE0067 Disposable object created by 'new DisposableThing()' is never disposed
        new DisposableThing(),
        // ↓ IDE0067 Disposable object created by 'new DisposableThing()' is never disposed
        new DisposableThing(),
        // No diagnostic, as expected
        new DisposableThing());
}

IDisposable CreateContainer(DisposableThing item1, DisposableThing item2, DisposableThing item3)
{
    return new DisposableContainer(new[] { item1, item2, item3 });
}
```

**Comment by jnm2** [comment]
https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000-dispose-objects-before-losing-scope seems to also be incorrect, maybe via the same underlying mechanism.

**Comment by mavasani** [comment]
Seems to be a DFA issue related to array initializers.

**Comment by mavasani** [comment]
Needs us to move to newer FlowAnalysis utilities for the fix to flow into Roslyn.
